Libraries/src/Amazon.Lambda.AspNetCoreServer.Hosting/ServiceCollectionExtensions.cs

```diff
@@ -88,6 +88,58 @@ public static IServiceCollection AddAWSLambdaHosting(this IServiceCollection ser
             return services;
         }
 
+        #if NET8_0_OR_GREATER
+        /// <summary>
+        /// Adds a <see cref="HttpRequestMessage"/>> that will be used to invoke
+        /// Routes in your lambda function in order to initialize the asp.net and lambda pipelines
+        /// during <see cref="SnapshotRestore.RegisterBeforeSnapshot"/>.  This improves the performance gains offered by SnapStart.
+        /// <para />
+        /// The returned <see cref="HttpRequestMessage"/> must have a relative
+        /// <see cref="HttpRequestMessage.RequestUri"/>.
+        /// <para />.
+        /// Be aware that this will invoke your applications function handler code
+        /// multiple times so that .NET runtime sees this code is a hot path and should be optimized.
+        /// <para />
+        /// When the function handler is called as part of SnapStart warm up, the instance will use a
+        /// mock <see cref="ILambdaContext"/>, which will not be fully populated.
+        /// <para />
+        /// This method automatically registers with <see cref="SnapshotRestore.RegisterBeforeSnapshot"/>.
+        /// <para />
+        /// This method can be called multiple times.
+        /// <para />
+        /// Example:
+        /// <para />
+        /// <code>
+        /// <![CDATA[
+        /// // Example Minimal Api
+        /// var builder = WebApplication.CreateSlimBuilder(args);
+        ///
+        /// builder.Services.AddAWSLambdaHosting(LambdaEventSource.HttpApi);
+        /// 
+        /// // Initialize asp.net pipeline before Snapshot
+        /// builder.Services.AddAWSLambdaBeforeSnapshotRequest(
+        ///     new HttpRequestMessage(HttpMethod.Get, "/test")
+        /// );
+        /// 
+        /// var app = builder.Build();
+        /// 
+        /// app.MapGet("/test", () => "Success");
+        /// 
+        /// app.Run(); 
+        /// ]]>
+        /// </code>
+        /// </summary>
+        public static IServiceCollection AddAWSLambdaBeforeSnapshotRequest(this IServiceCollection services, HttpRequestMessage beforeSnapStartRequests)
```

Comment:
nit: Remove the plural of `beforeSnapStartRequests` to `beforeSnapStartRequest` since it is a single request.

Reply:
done